reprojection.cpp

```diff
@@ -110,15 +111,12 @@ reprojection::~reprojection()
     pj_free(pj_tile);
     pj_tile = nullptr;
 
-    if (custom_projection != nullptr) {
-        delete custom_projection;
-    }
 }
 
-struct Projection_Info const *reprojection::project_getprojinfo(void)
+const std::shared_ptr<const Projection_Info> reprojection::project_getprojinfo(void)
 {
   if( Proj >= 0 )
-    return &Projection_Infos[Proj];
+    return std::make_shared<const Projection_Info>(Projection_Infos[Proj]);
```

Comment:
You are wrapping a pointer to static data in a shared pointer here. That's bad as it will call delete on it as soon as the pointer goes out of scope. shared_ptr is overkill here anyways as the projection info is never really shared with objects (the pointer is never saved anywhere else). Use a unique_ptr below and just return a raw pointer here as before. Or even better: replace project_getprojinfo() with a get_srs() and get_proj_desc() function.